### ISSUE TITLE
[d3] Parallelsm-dependent singular matrix error in cartesian runs

### ISSUE THREAD

**Issue by bpbrown** [DedalusProject/dedalus#136]
When running our `d3/examples/test_new_rbc2d.py` script from the tip of current `d3`, there appears to be a parallelism dependent singular matrix error.  This 2-d cartesian script runs fine on a single processor (`mpirun -n 1 python3 test_new_rbc2d.py`), but when two or more processors are used, the following error is thrown by all of the cores except one:

```
mpirun -n 2 python3 test_new_rbc2d.py
Problem built
Problem built
2021-04-30 14:37:15,936 subsystems 0/2 INFO :: Building subproblem matrices 1/64 (~2%) Elapsed: 0s, Remaining: 19s, Rate: 3.3e+00/s
2021-04-30 14:37:17,200 subsystems 0/2 INFO :: Building subproblem matrices 7/64 (~11%) Elapsed: 2s, Remaining: 13s, Rate: 4.5e+00/s
2021-04-30 14:37:18,673 subsystems 0/2 INFO :: Building subproblem matrices 14/64 (~22%) Elapsed: 3s, Remaining: 11s, Rate: 4.6e+00/s
2021-04-30 14:37:20,145 subsystems 0/2 INFO :: Building subproblem matrices 21/64 (~33%) Elapsed: 5s, Remaining: 9s, Rate: 4.7e+00/s
2021-04-30 14:37:21,635 subsystems 0/2 INFO :: Building subproblem matrices 28/64 (~44%) Elapsed: 6s, Remaining: 8s, Rate: 4.7e+00/s
2021-04-30 14:37:23,116 subsystems 0/2 INFO :: Building subproblem matrices 35/64 (~55%) Elapsed: 7s, Remaining: 6s, Rate: 4.7e+00/s
2021-04-30 14:37:24,593 subsystems 0/2 INFO :: Building subproblem matrices 42/64 (~66%) Elapsed: 9s, Remaining: 5s, Rate: 4.7e+00/s
2021-04-30 14:37:25,646 subsystems 0/2 INFO :: Building subproblem matrices 47/64 (~73%) Elapsed: 10s, Remaining: 4s, Rate: 4.7e+00/s
2021-04-30 14:37:26,069 subsystems 0/2 INFO :: Building subproblem matrices 49/64 (~77%) Elapsed: 10s, Remaining: 3s, Rate: 4.7e+00/s
2021-04-30 14:37:27,551 subsystems 0/2 INFO :: Building subproblem matrices 56/64 (~88%) Elapsed: 12s, Remaining: 2s, Rate: 4.7e+00/s
2021-04-30 14:37:29,019 subsystems 0/2 INFO :: Building subproblem matrices 63/64 (~98%) Elapsed: 13s, Remaining: 0s, Rate: 4.7e+00/s
2021-04-30 14:37:29,230 subsystems 0/2 INFO :: Building subproblem matrices 64/64 (~100%) Elapsed: 14s, Remaining: 0s, Rate: 4.7e+00/s
Iteration: 0
Iteration: 0
Traceback (most recent call last):
  File "test_new_rbc2d.py", line 112, in <module>
    solver.step(timestep)
  File "/Users/bpbrown/conda_install/src/dedalus-d3/dedalus/core/solvers.py", line 472, in step
    self.timestepper.step(dt, wall_time)
  File "/Users/bpbrown/conda_install/src/dedalus-d3/dedalus/core/timesteppers.py", line 624, in step
    sp.LHS_solvers[i] = solver.matsolver(sp.LHS, solver)
  File "/Users/bpbrown/conda_install/src/dedalus-d3/dedalus/libraries/matsolvers.py", line 121, in __init__
    self.LU = spla.splu(matrix.T.tocsc(), permc_spec='NATURAL')
  File "/Users/bpbrown/miniconda3/envs/dedalus-d3/lib/python3.8/site-packages/scipy/sparse/linalg/dsolve/linsolve.py", line 337, in splu
    return _superlu.gstrf(N, A.nnz, A.data, A.indices, A.indptr,
RuntimeError: Factor is exactly singular
```

**Comment by kburns** [comment]
Found the issue -- its a problem with chunks vs groups in Fourier distributions.  I'm working on a fix that should clean a lot of this up.

**Comment by bpbrown** [comment]
Awesome Keaton.  I just manufactured a LBVP test problem that works in serial and fails in parallel.

**Comment by kburns** [comment]
Should be fixed with #140.

**Comment by kburns** [comment]
Closing as I think this is resolved.
